card-games-vite/src/components/App/App.jsx

```diff
@@ -0,0 +1,394 @@
+          handleCloseModal();
+          resetForm();
+        }
+      })
+      .catch(console.error);
+  };
+
+  const handleRegistration = ({ email, password, name, avatar }, resetForm) => {
+    register(email, password, name, avatar)
+      .then((data) => {
+        if (data.name) {
+          handleLogin({ email, password }, resetForm);
+          resetForm();
+          handleCloseModal();
+        }
+      })
+      .catch(console.error);
+  };
+
+  const handleLogout = () => {
+    setIsLoggedIn(false);
+    localStorage.removeItem("jwt");
+    setUserData({});
+  };
+
+  const handleEditProfile = ({ name, avatar }, resetForm) => {
+    editProfileInfo(name, avatar)
+      .then((profile) => {
+        setUserData({
+          name: profile.name,
+          avatar: profile.avatar,
+          email: profile.email,
+          id: profile.id,
+        });
+        setCurrentUser(profile);
+        resetForm();
+        handleCloseModal();
+      })
+      .catch((err) => console.error);
+  };
+
+  const setUser = (token) => {
+    checkToken(token)
+      .then((user) => {
+        setCurrentUser(user);
+        setIsLoggedIn(true);
+      })
+      .catch((err) => {
+        console.error(err);
+        setIsLoggedIn(false);
+      });
+  };
+
+  const handleCardLike = (game) => {
+    const setGameCardLikes = (updatedGame) => {
+      setGameInfo((games) => {
+        return games.map((item) =>
+          item.id === updatedGame.id ? updatedGame : item
+        );
+      });
+    };
+
+    setIsLoading(true);
+    if (game.liked) {
+      dislikeGame(game.id, localStorage.getItem("jwt"))
+        .then((res) => {
+          game.liked = res.liked;
+          setGameCardLikes(game);
+        })
+        .catch(console.error)
+        .finally(setIsLoading(false));
+    } else {
+      likeGame(game.id, localStorage.getItem("jwt"))
+        .then((res) => {
+          game.liked = res.liked;
+          setGameCardLikes(game);
+        })
+        .catch(console.error)
+        .finally(setIsLoading(false));
+    }
+  };
+
+  const handleGameStart = (numberOfDecks) => {
+    setGameActive(true);
+    setIsLoading(true);
+    if (localStorage.getItem("deck_id") === null) {
+      createNewDeck(numberOfDecks)
+        .then((data) => {
+          localStorage.setItem("deck_id", `${data.deck_id}`);
+        })
+        .catch(console.error)
+        .finally(() => setIsLoading(false));
+    } else {
+      shuffle();
+    }
+  };
+
+  const handleGameEnd = () => {
+    setGameActive(false);
+    shuffle();
+  };
+
+  const shuffle = () => {
+    setIsLoading(true);
+    shuffleAllCards(localStorage.getItem("deck_id"))
+      .then((data) => {
+        if (data.success) {
+          setHand([]);
+          setDiscardPile([]);
+          renderDrawPile(data.remaining);
+        } else {
+          console.log("Sorry, an error has occurred");
+        }
+      })
+      .catch(console.error)
+      .finally(setIsLoading(false));
```

Comment:
In `then`, `catch`, `finally` you need to pass only the **reference** to the function (not a function call) otherwise the function is called immediately without waiting for the response from the server. You can pass an **unnamed** arrow function as well.